### DISCUSSION TITLE
Custom JWT sessions

### DISCUSSION THREAD

**Discussion by zane-atomus** [supabase/supabase#28073] [Questions] [answered]
I have successfully implemented exchanging an external JWT for a Supabase JWT using the Supabase JWT secret. From there I send the token back to the browser and create a Supabase client with that token in global headers. This all works great, but ideally I would like to create a session in Supabase to take advantage of Supabase's built in session management. Is there a way to accomplish this, or does that only happen if you use one of the `supabase.auth.signIn<...>` methods?

**Comment by GaryAustin1** [comment, marked as answer]
You have to use a sign in method.
You might also be able to use an admin generatelink  https://supabase.com/docs/reference/javascript/auth-admin-generatelink from server side to get a token and then call verifyOtp also on the server to get a session (access token and refresh token).  I've not personally tried it though.

**Reply by zane-atomus** [reply]
Good idea, that will work for us at the moment. Thanks!
